### ISSUE TITLE
Translation error in German roombooking error message when trying to book an already booked timeslot

### ISSUE THREAD

**Issue by moschn-unibonn** [indico/indico#6469]
**Describe the bug**
See issue title, not much else to say about it.

**To Reproduce**
1. Select a room (e.g. from the bookings overview)
2. Start a booking and select a timeslot which is already booked
3. An error message is displayed: "Raum kann nicht reserviert werden. Der Raum ist in dem/den ausgewählten Zeitfenster(n) frei"

**Expected behavior**
The second sentence of the translation seems to be missing a "nicht" or other negation to indicate that the room is in fact not free to book at the selected timeslot.
(The English error message is: "Space cannot be booked. One or more bookings would conflict with yours.")

**Comment by ThiefMaster** [comment]
Those are different messages, both in the English source and the German translation. Are you sure the conditions when testing in German/English were exactly the same?

FYI this is the code containing those strings:

```jsx
function ConflictIndicator({availability: {numDaysAvailable, allDaysAvailable}}) {
  // todo: warning in case there are pre-booking conflicts
  return allDaysAvailable ? (
    <Message color="green">
      <Icon name="check" />
      <Translate>The space will be free on the selected time slot(s)</Translate>
    </Message>
  ) : numDaysAvailable ? (
    <Message color="yellow">
      <Icon name="warning sign" />
      <Translate>The space won't be available on one or more days</Translate>
    </Message>
  ) : (
    <Message color="red">
      <Message.Header>
        <Icon name="remove" />
        <Translate>Space cannot be booked.</Translate>
      </Message.Header>
      <Translate>One or more bookings would conflict with yours.</Translate>
    </Message>
  );
}
```

**Comment by moschn-unibonn** [comment]
Yes, I'm sure. Following the steps above in English produces the "One or more bookings would conflict with yours." message.
In German it displays like this:
![roombooking-german-error](https://github.com/user-attachments/assets/2a4a8d6e-d253-4eed-b4f7-abb93d85a828)
Both times selecting the same date and timeslot, as a single booking (if that is important).

**Comment by ThiefMaster** [comment]
Is there any chance that you are using a different date when testing w/ english? 08.09. vs 09.08. would be my guess.

Because the translation strings look fine:

```
really really strange, because the translation dictionary shows this:

```
#: indico/modules/rb/client/js/modules/bookRoom/BookFromListModal.jsx:34
msgid "The space will be free on the selected time slot(s)"
msgstr "Der Raum ist in dem/den ausgewählten Zeitfenster(n) frei"

#: indico/modules/rb/client/js/modules/bookRoom/BookFromListModal.jsx:39
msgid "The space won't be available on one or more days"
msgstr "Der Raum ist an einem oder mehreren Tagen nicht verfügbar"

#: indico/modules/rb/client/js/modules/bookRoom/BookFromListModal.jsx:45
msgid "Space cannot be booked."
msgstr "Raum kann nicht reserviert werden."
```
```

**Comment by moschn-unibonn** [comment]
I just checked again with another booking on August 12, same result.

English (en_GB):
![roombooking-german-error-enGB](https://github.com/user-attachments/assets/8f3121b6-2e1b-4a8c-86a2-e8c4f3cd6e0d)

German:
![roombooking-german-error-deDE](https://github.com/user-attachments/assets/d0316ebc-9686-49ff-8918-b39135791496)

**Comment by ThiefMaster** [comment]
Which Indico version are you using? Any customizations made to the code?

In case it's https://indico.hiskp.uni-bonn.de (the only one I found when googling for "indico uni bonn"), then you are using an outdated version (v3.2.7) and should (ask your admin to) update to the latest version.

Even though the code of that version looks correct as well:

- https://github.com/indico/indico/blob/v3.2.7/indico/modules/rb/client/js/modules/bookRoom/BookFromListModal.jsx
- https://github.com/indico/indico/blob/v3.2.7/indico/translations/de_DE/LC_MESSAGES/messages-react.po#L4571-L4581

**Comment by moschn-unibonn** [comment]
No, this is a different indico and it's running the latest release 3.3.3, the error occurs on our test server as well (same release). We have only made some minor customizations in css and the header templates (changing colors, adding additional logos to the headers), nothing else. Language-wise we have disabled all languages except English (GB+US) and German via the indico.conf.

Are you able to reproduce the issue on your system?

**Comment by ThiefMaster** [comment]
In that case, please send me the path and JSON response of the request sent to `.../rooms/api/rooms/XXX/timeline?XXX` when that message is shown (use your browser dev tools to get it). Ideally both while the language is set to en_GB and while it's set to de_DE, just to see if there are any differences.

You could also use (at least in firefox) the "save as HAR" option on the request (in the network tab) which creates an archive containing the full request/response, but that one contains somewhat sensitive data such as the session cookie so if you go for that option, better email it to me (adrian.moennich@cern.ch) and logout and login again afterwards to invalidate the session cookie.

**Comment by olifre** [comment]
FWIW, I can also reproduce this with the CERN instance:
![grafik](https://github.com/user-attachments/assets/279f3fea-e6ca-4c00-a256-38090b4837aa)

**Comment by olifre** [comment]
> Which browser are you using? It's really strange because my dev instance running the same code as indico.cern.ch did not have the problem.

I tested with Firefox 128.0.3 and Google Chrome 126.0.6478.126 on Linux, both show the same problem. 

> Anyway, could you share the exact link from your browser's address bar when you had the dialog with the error message open?

Sure:
https://indico.cern.ch/rooms/rooms?modal=room-details%3A284&modal=book-room%3A284%3A%7B%22isPrebooking%22%3Atrue%7D

Testing further, I found that if I switch to French, I get:
```
L'espace ne peut pas être réservé.
L'espace sera disponible aux périodes sélectionnées
```
i.e. also a wrong message, so it seems it's not a problem of the German translation itself.

**Comment by ThiefMaster** [comment]
OK, I can reproduce it. Still an extremely weird bug!

**Comment by olifre** [comment]
Extremely weird indeed ;-).

**Comment by ThiefMaster** [comment]
Looks like it's a matter of React doing something weird and not properly refreshing the element - it first renders with `{numDaysAvailable: 1, allDaysAvailable: true}` and when changing the times to the slot that's unavailable, the component re-renders with `{numDaysAvailable: 0, allDaysAvailable: false}`, but for some reason it keeps the previous translated string.

**Comment by ThiefMaster** [comment]
This will be fixed in v3.3.4 (which I expect to release soon-ish, almost certainly this month).
